contrib/platform/src/com/sun/jna/platform/win32/WinCrypt.java

```diff
@@ -72,7 +72,11 @@ public DATA_BLOB(Pointer memory) {
         }
 
         public DATA_BLOB(byte [] data) {
-            pbData = new Memory(data.length);
+            // We allocate at least 1 byte memory region because `malloc` may return `NULL` if requested size is 0.
+            // But `CryptProtectData` and `CryptUnprotectData` consider `NULL` as invalid data (even the size is 0)
+            // It's acceptable to allocate 1 byte more and this does not affect the final result because
+            // we pass correct size explicitly on `cbData` field.
```

Comment:
If I read this comment correctly, you're saying a `null` pointer is invalid, so we need a valid pointer that we will ignore the contents of.  If this is the case, the subsequent `pbData.write()` call is useless.   I think rather than always invoking `Math.max()` this should be a simple conditional.  If `data.length` > 0 we create memory of that length and write to it; otherwise we can create any valid pointer allocation, e.g.,`new Memory(1)`.